base_delivery_carrier_label/models/delivery_carrier.py

```diff
@@ -28,6 +30,16 @@ def send_shipping(self, pickings):
         """Handle labels and  if we have them. Expected format is {'labels': [{}, ...]}
         The dicts are input for stock.picking#attach_label"""
         result = super().send_shipping(pickings)
+        # We could want to generate labels calling a method that does not depend
+        # on one specific delivery_type.
+        # For instance, if we want to generate a default label in case there are not
+        # carrier
+        # we may want to call another method not based on any delivery_type.
+        # or at the contrary, we could call a common method for multiple delivery_type
+        # for instance, in delivery_roulier, the same method is always called for any
+        # carrier implemented in roulier lib.
+        if result is None:
+            result = self.alternative_send_shipping(pickings)
```

Comment:
Does it deprecate https://github.com/OCA/delivery-carrier/tree/12.0/delivery_carrier_label_default ?

Reply:
No, it does not deprecate delivery_carrier_label_default module.
Actually, I don't use it but I guess it does not.
I had to add this `alternative_send_shipping` method so it is still possible to achieve what is done in `delivery_carrier_label_default` (and also in `delivery_roulier`).

Because, natively, Odoo will call a method `{carrier}_send_shipping()`, meaning we have to define one method by implemented carrier.
But, in `delivery_carrier_label_default` and in `delivery_roulier`, we want to implement one method that will work for multiple different carrier. In these cases, implementing something like `{carrier}_send_shipping()` is not possible. I added this `alternative_send_shipping` hook, so these modules can override it to implement their logic, that will work for multiple carriers.